core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java

```diff
@@ -89,12 +85,13 @@ public void sweep(String workflowId) {
                         "Workflow NOT found for id:{}. Removed it from decider queue",
                         workflowId,
                         e);
+                return;
             }
         } catch (Exception e) {
-            queueDAO.setUnackTimeout(
-                    DECIDER_QUEUE, workflowId, properties.getWorkflowOffsetTimeout().toMillis());
             Monitors.error(CLASS_NAME, "sweep");
             LOGGER.error("Error running sweep for " + workflowId, e);
         }
+        queueDAO.setUnackTimeout(
+                DECIDER_QUEUE, workflowId, properties.getWorkflowOffsetTimeout().toMillis());
```

Comment:
Before the change, if any exception occur during any step of this method, we will call setUnackTimeout` to extend lease. With the change, it is still seem to be the same thing. Did I miss anything? What is the bug that we are trying to fix?

Reply:
>Before the change, if any exception occur during any step of this method, we will call setUnackTimeout` to extend lease.

Actually, when an ApplicationException was thrown with a `Code != NOT_FOUND` the method setUnackTimeout was not invoked. So, in case of transient errors (Code == BACKEND_ERROR), the message would be removed from the decider queue. This change rectifies this behavior.